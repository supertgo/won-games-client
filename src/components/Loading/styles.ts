import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.main`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Text = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    color: ${theme.colors.white};
    margin: ${theme.spacings.xxsmall};
    animation: fade 1s linear 0s infinite;

    ${media.greaterThan('medium')`
    font-size: ${theme.font.sizes.xxlarge};
    `}
  `}

  @keyframes fade {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }
`;

export const Spinner = styled.div`
  ${({ theme }) => css`
    width: ${theme.spacings.medium};
    height: ${theme.spacings.medium};
    border: 3px solid ${theme.colors.primary};
    border-top: 3px solid transparent;
    border-radius: 50%;
    animation: spin 1s linear 0s infinite;

    ${media.greaterThan('medium')`
      width: ${theme.spacings.xlarge};
      height: ${theme.spacings.xlarge};
    `}
  `}

  @keyframes spin {
    from {
      transform: rotate(0);
    }

    to {
      transform: rotate(360deg);
    }
  }
`;
