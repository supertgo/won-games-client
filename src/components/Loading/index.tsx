import * as S from './styles';

const Loading = () => (
  <S.Wrapper>
    <S.Text>Loading...</S.Text>
    <S.Spinner />
  </S.Wrapper>
);

export default Loading;
