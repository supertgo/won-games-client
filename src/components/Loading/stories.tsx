import { Story, Meta } from '@storybook/react';
import Loading from '.';

export default {
  title: 'Loading',
  component: Loading,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  }
} as Meta;

export const Default: Story = () => <Loading />;
