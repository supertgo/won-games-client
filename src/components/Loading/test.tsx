import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Loading from '.';

describe('<Loading />', () => {
  it('should render the loading component', () => {
    const { container } = renderWithTheme(<Loading />);

    expect(screen.getByText(/loading/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
