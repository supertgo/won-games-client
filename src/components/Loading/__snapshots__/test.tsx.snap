// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the loading component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.0rem;
  color: #FAFAFA;
  margin: 0.8rem;
  -webkit-animation: fade 1s linear 0s infinite;
  animation: fade 1s linear 0s infinite;
}

.c2 {
  width: 3.2rem;
  height: 3.2rem;
  border: 3px solid #F231A5;
  border-top: 3px solid transparent;
  border-radius: 50%;
  -webkit-animation: spin 1s linear 0s infinite;
  animation: spin 1s linear 0s infinite;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 4.8rem;
    height: 4.8rem;
  }
}

<main
  class="c0"
>
  <span
    class="c1"
  >
    Loading...
  </span>
  <div
    class="c2"
  />
</main>
`;
